src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V62/Messages/BlockBodiesMessageSerializer.cs

```diff
@@ -10,20 +10,43 @@ namespace Nethermind.Network.P2P.Subprotocols.Eth.V62.Messages
 {
     public class BlockBodiesMessageSerializer : IZeroInnerMessageSerializer<BlockBodiesMessage>
     {
-        public byte[] Serialize(BlockBodiesMessage message)
+        private readonly TxDecoder _txDecoder = new();
+        private readonly HeaderDecoder _headerDecoder = new();
+
+        public void Serialize(IByteBuffer byteBuffer, BlockBodiesMessage message)
         {
-            return Rlp.Encode(message.Bodies.Select(b => b is null
-                ? Rlp.OfEmptySequence
-                : Rlp.Encode(
-                    Rlp.Encode(b.Transactions),
-                    Rlp.Encode(b.Uncles))).ToArray()).Bytes;
+
+            int totalLength = GetLength(message, out int contentLength);
```

Comment:
Very common pattern. Maybe one day we can DRY this out somehow.

Reply:
yes, I plan to do this next.
there are multiple different patterns for getLength used across the solution. We can create a proper spec for this.